store/sqlstore/file_info_store.go

```diff
@@ -95,6 +96,72 @@ func (fs SqlFileInfoStore) Get(id string) (*model.FileInfo, *model.AppError) {
 	return info, nil
 }
 
+func (fs SqlFileInfoStore) GetWithOptions(opt *model.GetFilesOptions) ([]*model.FileInfo, *model.AppError) {
+	if opt == nil {
+		opt = &model.GetFilesOptions{}
+	}
```

Comment:
`nil` options is valid because the zero value of `model.GetFilesOptions` is valid. It gets all files except the deleted ones. I think is a reasonable default.

I code does some basic validity checks of the options passed in, but it might better to have a method to check if the options are valid.